### PR TITLE
Fix Vector info bug #359

### PR DESCRIPTION
**Pull request by linakrisztian**
#359: By changing v.info -c to v.db.connect -c, GRASS does not return an error for vector maps with missing table. Instead this message is printed: "Map <test_random_vect> is not connected to a database". Thus the endpoint does not return an error anymore.

### CODE REVIEW

**Review by griembauer — APPROVED** [review]
Thanks, looks good and tested successfully locally!

File: src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def _execute(self):
             "flags": "h"}
 
         pc["3"] = {
-            "module": "v.info",
+            "module": "v.db.connect",
             "inputs": {"map": vector_name + "@" + self.mapset_name},
             "flags": "c"}
```
Comment (metzm) [inline src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py:70]: Do you really want to print types/names of table columns for the specified layer (default is layer 1) or rather print all map connection parameters for all layers, would be empty if there are no connections?

```suggestion
            "flags": "g"}
```

Comment (linakrisztian) [reply src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py:70]: @anikaweinmann or @mmacata : The old case (v.info -c) also printed only type/names of the specified layer (default layer 1). Should we change it as @metzm suggested?

**Comment by metzm** [comment]
A GRASS vector can have no database connection at all or one or several database connections, one possible connection for each layer. The common case is that a vector has one database connection, the corresponding layer is layer 1. For other cases `v.info -c` or `v.db.connect -c` does not make sense because no columns would be reported or only columns for layer 1, ignoring attribute columns for other layers.
The class `EphemeralVectorInfo()`reports general information about a vector. Therefore I would argue that this is the wrong place for column names and types of a specific layer. Instead, any database connections as reported by `v.db.connect -g()` would be useful information in this place. After that, a user might want to know column names and types of a specific vector layer, see also https://github.com/OSGeo/grass/blob/main/python/grass/script/vector.py#L101.

Maybe introduce a new endpoint where the layer of a vector can be specified?

**Comment by metzm** [comment]
Instead of a new endpoint for a vector layer, the syntax also recognized by TGIS could be used: `<name>:<layer>@<mapset>`. If a layer is given, `v.db.connect -c layer=<layer> map=<name>` could be used, otherwise `v. db.connect -g map=<name>`.

File: src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py (hunk 1 above)
Comment (anikaweinmann) [reply src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py:70]: I think `v.db.connect -c` is the one which is like it is now. So I would prefer this. The style of `g` is anonther one than we have a breaking change, or not?

Comment (anikaweinmann) [reply src/actinia_core/processing/actinia_processing/ephemeral/vector_layer.py:70]: Check with `v.db.connect -g` if there are more than 1 layer/tables or no layers (count number of rows). If there is only one layer the output should be `v.info -c ` as before, by multiple layer/tables `v.db.connect -g` should be inside the results. By no layer/table nothing has to be added to the results.

Open question: how to write this in the api docs.

**Review by metzm — APPROVED** [review]

**Pull request merged**